cmd/crictl/container.go

```diff
@@ -432,13 +446,19 @@ func CreateContainer(client pb.RuntimeServiceClient, opts createOptions) (string
 		}
 	}
 
+	// Try to pull the image before container creation
+	image := config.GetImage().GetImage()
+	if _, err := PullImage(iClient, image, nil); err != nil {
```

Comment:
I like the idea, but if local image is used, the pull operation would fail depending on the runtime implementations.

So, I think we'd better add an option here, so that the pull can be skipped in some cases?

Reply:
Yes, I agree. I added the option to `no-pull` as well as the pull authentication related options (this should work too I guess).

Reply:
thanks